### ISSUE TITLE
Book travel - Book travel animation becomes blank while RHP is dismissed

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#49744]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.40-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** applausetester+kh230901@applause.expensifail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to staging.new.expensify.com
2. Open FAB
3. Click Book travel
4. Dismiss the RHP

## Expected Result:
Book travel animation will not become blank while RHP is dismissed

## Actual Result:
Book travel animation becomes blank while RHP is dismissed

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/962378ab-7d61-4d27-90c2-77675bb02f37

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by rushatgabhane** [comment]
don't know what caused it, but def NOT A BLOCKER

**Comment by MonilBhavsar** [comment]
Thanks for looking! Demoting from blocker, but let's try if we can fix it

**Comment by rushatgabhane** [comment]
@MonilBhavsar @neonbhai the reasons mentioned here suggest that we should do nothing. Or it may cause issues 1, 2, 3

https://github.com/Expensify/App/blob/a4a344b7cedd0ab9cafb4857db15de8eea409689/src/components/Lottie/index.tsx#L82-L88

**Comment by MonilBhavsar** [comment]
Since this was reported as a regression, may be it comes from https://github.com/Expensify/App/pull/48040 🤔 
Going to check

**Comment by MonilBhavsar** [comment]
Not reproducible after reverting https://github.com/Expensify/App/pull/48040
@Krishna2323 @hungvu193 if you could please take a look

**Comment by hungvu193** [comment]
> Not reproducible after reverting #48040 @Krishna2323 @hungvu193 if you could please take a look

No. It came from this one (https://github.com/Expensify/App/pull/48444). It shows empty view while navigating away.
https://github.com/Expensify/App/blob/33818a904e9db22d3e06feaab8b5419993f322bd/src/components/Lottie/index.tsx#L88

Long story short, While working on  #48040 with @Krishna2323, we found a bug, and it from PR of https://github.com/Expensify/App/issues/47273, I commented there https://github.com/Expensify/App/issues/47273#issuecomment-2317717050. After that, the contributor of https://github.com/Expensify/App/issues/47273 submitted another PR to fix the issue that we found which is https://github.com/Expensify/App/pull/48444

**Comment by MonilBhavsar** [comment]
https://github.com/Expensify/App/pull/48444 was deployed to production 2 weeks ago and this bug was only reproducible on staging. I don't think so it has caused this, unless I am missing something. 

The bug stems from one of the PR from checklist https://github.com/Expensify/App/issues/49662

**Comment by hungvu193** [comment]
> #48444 was deployed to production 2 weeks ago and this bug was only reproducible on staging. I don't think so it has caused this, unless I am missing something

It's because we used static images for the BookTravel placeholder before. We switched to use LottieAnimation recently that's why you saw this issue. That being said, this issue will happen with any RHP that has Lottie animation because we hide them while navigating away to fix https://github.com/Expensify/App/issues/47273
cc @QichenZhu and @suneox

**Comment by QichenZhu** [comment]
@hungvu193 Previously, playing animations in the background made the user interface unresponsive (#47273), so we stopped them after navigating away.

Now, after some updates in the app (possibly the RN upgrade), performance has improved. Although background animations still consume CPU, they no longer make the interface unresponsive.

If the high CPU usage is acceptable, we could allow animations to play in the background again.

cc @suneox

**Comment by suneox** [comment]
> I commented there [#47273 (comment)](https://github.com/Expensify/App/issues/47273#issuecomment-2317717050). 

@hungvu193 I have another comment to verify behavior on [PR](https://github.com/Expensify/App/pull/47960) still works as expected after your comment

> After that, the contributor of https://github.com/Expensify/App/issues/47273 submitted another PR to fix the issue that we found

Nope, the PR is fix crashed [issue](https://github.com/Expensify/App/issues/48262) 

> This issue will happen with any RHP that has Lottie animation because we hide them while navigating away to fix https://github.com/Expensify/App/issues/47273

Before this [PR](https://github.com/Expensify/App/pull/48040), we didn’t have any use cases similar to this issue.

> https://github.com/Expensify/App/pull/48444 was deployed to production 2 weeks ago and this bug was only reproducible on staging. I don't think so it has caused this, unless I am missing something.

@MonilBhavsar  Actually, I don’t think the PR from 2 weeks ago is a regression of the new feature because, at the time of the old PR, we didn’t have any context for the new feature. So, I believe this should be considered a new issue, and the scope of the old PR doesn’t cover the new one.

**Comment by suneox** [comment]
@QichenZhu  I believe our PR shouldn’t cover the new feature introduced several weeks later, so you can go provide a new proposal for this issue.​

**Comment by hungvu193** [comment]
@suneox  Hey I'm not trying to be picky here. This issue is clearly a regression from https://github.com/Expensify/App/pull/48444. 
We can reproduce this issue even when we reverted https://github.com/Expensify/App/pull/48040. It happened with every single Lottie animation, but it can be easily reproducible with RHP.

https://github.com/user-attachments/assets/5b6d4c08-d953-4b67-aac4-f4273de738cd

https://github.com/user-attachments/assets/dd58121c-0308-4aa4-a22b-23fa91b6d8d5


That means, QA didn't find it during regression period of https://github.com/Expensify/App/pull/48444. Since regression period was over for https://github.com/Expensify/App/pull/48444. We can consider this as a new issue.

**Comment by QichenZhu** [comment]
@hungvu193 Yes, you're right that similar issues exist on other pages too. It wasn’t considered a problem before, probably because it’s not noticeable to the human eye. I have to use a slow motion camera to catch it.

Why is this issue noticeable on the travel page? Because the page slides out more softly. Below is a comparison video of the Preference and Travel pages at 8x slower speed.

https://github.com/user-attachments/assets/648c9b4c-0234-4abb-9862-a026ca82816f

**Comment by suneox** [comment]
> it’s not noticeable to the human eye, I have to use a slow motion camera to catch it.

Agreed, I only caught the issue by recording and replaying it on a different page.  So I think it should be handled as a new issue

@neonbhai [proposal](https://github.com/Expensify/App/issues/49744#issuecomment-2375270022) is straightforward for fixing the issue.

**Comment by MonilBhavsar** [comment]
Okay @suneox assigning you to the issue as a C+

**Comment by MonilBhavsar** [comment]
@neonbhai mind providing the ETA when you have a moment please

**Comment by QichenZhu** [comment]
> pause the lottie animation view while navigating away

@layacat Just a little heads up, the animation needs to resume playing after navigating back.

**Comment by suneox** [comment]
@MonilBhavsar We have another solution that also resolves this issue.

* @neonbhai [proposal](https://github.com/Expensify/App/issues/49744#issuecomment-2375270022) is to delay setting the navigation state to away.
* @layacat [proposal](https://github.com/Expensify/App/issues/49744#issuecomment-2387822904) is to trigger a pause in animation when the state change to away.

Both proposals seem to resolve the issue, but @layacat approach looks better than @neonbhai approach, so Should we consider changing the current selected proposal to @layacat [proposal](https://github.com/Expensify/App/issues/49744#issuecomment-2387822904) at this time?

🎀 👀 🎀 C+ reviewed

**Comment by sakluger** [comment]
Summarizing payment on this issue:

Contributor: @layacat $250, **please apply to the [Upwork job](https://www.upwork.com/jobs/~021840763606281462254) or comment with your Upwork profile**
Contributor+: @suneox $250, paid via Upwork

@suneox please complete the BZ checklist.

**Comment by suneox** [comment]
### Checklist

- [x] [@suneox] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/47960
- [x] [@suneox] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. [Link to comment](https://github.com/Expensify/App/pull/47960/files#r1806077240) 
- [x] [@suneox] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: *N/A*
- [x] [@suneox] Determine if we should create a regression test for this bug: *N/A*
- [x] [@suneox] If we decide to create a regression test for the bug, please propose the regression test steps to ensure the same bug will not reach production again. *N/A*

**Comment by sakluger** [comment]
@layacat thanks! I didn't realize you had already applied. I sent an offer via Upwork.

**Comment by sakluger** [comment]
Perfect, thanks! All contracts have been paid out now.
